### Eclipse Workspace Patch 1.0
#P ao.dev
Index: sites/all/modules/changed/model/model.module
===================================================================
--- sites/all/modules/changed/model/model.module	(revision 1514)
+++ sites/all/modules/changed/model/model.module	(working copy)
@@ -146,6 +146,11 @@
  *   Whether access is allowed or not.
  */
 function model_access($op, $model = NULL, $account = NULL) {
+	if (empty($account)) {
+	  global $user;
+	  $account = $user;
+	}
+	
   if (user_access('administer models', $account)) {
     return TRUE;
   }
Index: sites/all/modules/changed/model/model.admin.inc
===================================================================
--- sites/all/modules/changed/model/model.admin.inc	(revision 1514)
+++ sites/all/modules/changed/model/model.admin.inc	(working copy)
@@ -57,7 +57,7 @@
         'page callback' => 'model_form_wrapper',
         'page arguments' => array(model_create(array('type' => $type->type))),
         'access callback' => 'model_access',
-        'access arguments' => array('edit', 'edit ' . $type->type),
+        'access arguments' => array('edit', $type),
         'file' => 'model.admin.inc',
         'file path' => drupal_get_path('module', $this->entityInfo['module'])
       );

